cmd/snap-seccomp/main.go

```diff
@@ -466,13 +466,60 @@ func (sc *SeccompData) SetArgs(args [6]uint64) {
 	}
 }
 
-func readNumber(token string) (uint64, error) {
+// Only support negative args for syscalls where we understand the glibc/kernel
+// prototypes and behavior. This lists all the syscalls that support negative
+// arguments and where we want to ignore the high 32 bits (ie, we'll masq it)
+// since the arg is known to be 32 bit (uid_t/gid_t) and the kernel accepts one
+// or both of uint32(-1) and uint64(-1) and does its own masking.
+var syscallsWithNegArgsMasqHi32 = map[string]bool{
```

Comment:
I'm not sure I fully understood  fc95f8e  the two sets were not the same but one a subset of the other

Reply:
@pedronis - at first I was going to try to handle negative arguments to syscalls generically such that you could specify a negative value for any syscall in our policy, but as I went deeper into the subject, I found that we need to look at each individual syscall independently to understand the interplay between the kernel, libseccomp/golang-seccomp and glibc (remembering the ioctl issue). I found that different syscalls behaved differently and decided instead of having two lists where one lists anything that accepts a negative arg and another that tries to give a hint as to when to mask, we would instead have one list with the behavior we understand and consistent policy generation for that list (all the syscalls I was interested in for this PR behaved the same on further inspection, but when looking at other syscalls that took negative args some, for example, setpriority, did not). If we want to expand our policy language to support differently behaving syscalls, we'll create a new list (I know of none at this time).

I did quite a bit of rebasing in the PR and admit the commit message did not capture this.

Reply:
I should also mention, there aren't a lot of syscalls that take a negative argument, and between chown and setuid/gid families of syscalls, and setpriority that we handle differently, we've hit the major ones already.